### PR TITLE
Automate release push to OLM in tempo-operator

### PR DESCRIPTION
**Pull request by rubenvp8510**
It will be good to merge first https://github.com/grafana/tempo-operator/pull/681

Also I'm not sure how to test this before release

### CODE REVIEW

**Comment by iblancasa** [comment]
> Also I'm not sure how to test this before release

@rubenvp8510 I think you can do it from your fork (you will need to merge first to `main` in your fork).

**Review by iblancasa — APPROVED** [review]

File: .github/workflows/reusable-operator-hub-release.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          # (fetching the upstream repo with "--depth 1" would lead to "shallow update not allowed"
+          #  error when pushing back to the origin repo)
+          gh repo sync grafana/${{ inputs.repo }} \
+              --source ${{ inputs.org }}/${{ inputs.repo }} \
+              --force
+
+      - name: Checkout operatorhub repo
+        uses: actions/checkout@v3
+        with:
+          repository: grafana/${{ inputs.repo }}
```
Comment (pavolloffay) [inline .github/workflows/reusable-operator-hub-release.yaml:45]: this looks fine

https://github.com/grafana/community-operators
https://github.com/grafana/community-operators-prod

File: .github/workflows/reusable-operator-hub-release.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+jobs:
+  create-operator-pull-request:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
```
Comment (pavolloffay) [inline .github/workflows/reusable-operator-hub-release.yaml:21]: this should be PUBLISH_APP_ID

File: .github/workflows/reusable-operator-hub-release.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+          private_key: ${{ secrets.PRIVATE_KEY }}
```
Comment (pavolloffay) [inline .github/workflows/reusable-operator-hub-release.yaml:22]: this should be PUBLISH_APP_PRIVATE_KEY

File: .github/workflows/reusable-operator-hub-release.yaml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        required: true
+
+jobs:
+  create-operator-pull-request:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
```
Comment (pavolloffay) [inline .github/workflows/reusable-operator-hub-release.yaml:20]: we should as well set  installation_id: PUBLISH_APP_INSTALLATION_ID

Comment (rubenvp8510) [reply .github/workflows/reusable-operator-hub-release.yaml:20]: Where did you get `PUBLISH_APP_INSTALLATION_ID`, I don't see where is mentioned on the documentation but I maybe missing something.

**Comment by rubenvp8510** [comment]
This should work in theory

In the practice I'm getting this error:

`pull request create failed: GraphQL: Resource not accessible by integration (createPullRequest)`

Still don't know what it means, the example App I used has all the permissions.

File: .github/workflows/reusable-operator-hub-release.yaml (hunk 4 above)
Comment (rubenvp8510) [reply .github/workflows/reusable-operator-hub-release.yaml:27]: ping?

Comment (pavolloffay) [reply .github/workflows/reusable-operator-hub-release.yaml:27]: Correct it does not seem to be mentioned in https://github.com/tibdex/github-app-token, maybe perhaps as `installation_retrieval_mode`, but probably not. 

The PUBLISH_APP_INSTALLATION_ID is defined as a repository secret, but perhaps we don't have to use it.

**Review by iblancasa — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Review by frzifus — APPROVED** [review]

**Comment by rubenvp8510** [comment]
This is not gonna work, as the installation tokens only have read-only access to the public repos, means no PR creation. we need a PAT, I'm modifying this to make it work as same way OTel does.

File: .github/workflows/reusable-operator-hub-release.yaml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Sync fork
+        env:
+          GH_TOKEN: ${{ secrets.TEMPOOPERATORBOT_GITHUB_TOKEN }}
+        run: |
+          # synchronizing the fork is fast, and avoids the need to fetch the full upstream repo
+          # (fetching the upstream repo with "--depth 1" would lead to "shallow update not allowed"
+          #  error when pushing back to the origin repo)
+          gh repo sync rubenvp8510/${{ inputs.repo }} \
```
Comment (andreasgerstmayr) [inline .github/workflows/reusable-operator-hub-release.yaml:29]: This command has your hardcoded username in it

Comment (rubenvp8510) [reply .github/workflows/reusable-operator-hub-release.yaml]: good catch! It was for testing.

**Comment by rubenvp8510** [comment]
This is ready, the only requirement is to add the PAT to the `TEMPOOPERATORBOT_GITHUB_TOKEN` secret at repository level.

**Pull request merged**